crates/primitives/src/chain/spec.rs

```diff
@@ -574,6 +574,11 @@ impl ChainSpec {
     /// Returns `true` if this chain contains Optimism configuration.
     #[inline]
     pub fn is_optimism(&self) -> bool {
+        #[cfg(feature = "optimism")]
+        if self.hardforks.contains_key(&Hardfork::Bedrock) {
+            return true;
+        };
```

Comment:
this seems reasonable, doesn't impact regular eth nodes

Reply:
I considered simply returning `false` if the `optimism` feature is disabled. Is it worth also making this change or simply keeping the previous name check?

Reply:
we are trying to get rid of the feature all together, so this is good the way it is https://github.com/paradigmxyz/reth/issues/7649